src/scss/components/checkbox/_index.scss

```diff
@@ -142,120 +121,73 @@ hx-checkbox-control {
 
     &:checked,
     &:indeterminate {
-      color: @cyan-900;
+      color: $cyan-900;
 
       ~ label > hx-checkbox {
-        border-color: @cyan-900;
-        color: @cyan-900;
+        border-color: $cyan-900;
+        color: $cyan-900;
       }
     }
 
     &:valid {
       ~ label > hx-checkbox:hover {
-        background-color: @cyan-50;
-        border-color: @cyan-500;
-        color: @cyan-500;
+        @include hxCheckboxFacade($pseudo-state: hover);
       }
     }
   }
 }
 
 /* ========== INVALID ========== */
-.isInvalid() {
-  color: @red-900;
-
-  &::-ms-check {
-    border-width: 2px;
-    border-color: @red-900 !important;
-    color: @red-900 !important;
-  }
+hx-checkbox-control[hx-dirty] > input[type="checkbox"]:invalid,
+hx-checkbox-control.hxInvalid > input[type="checkbox"]:enabled {
+  @include hxCheckboxControl;
 
   &:focus {
-    box-shadow: @focus-glow-invalid;
-
-    &::-ms-check {
-      border-color: @red-900;
-    }
+    @include hxCheckboxControl(invalid, focus);
 
     ~ label > hx-checkbox {
-      border-color: @red-900;
-      box-shadow: @focus-glow-invalid;
+      @include hxCheckboxFacade(invalid, focus);
     }
   }
 
   &:hover {
-    &::-ms-check {
-      background-color: @red-100;
-    }
+    @include hxCheckboxControl(invalid, hover);
   }
 
   ~ label > hx-checkbox {
-    border: 2px solid @red-900;
-    color: @red-900;
+    @include hxCheckboxFacade;
```

Comment:
Should this be invalid, instead of pristine?

```scss
@include hxCheckboxFacade(invalid);
```

Reply:
Updated mixin `$state`.